Compile android template error on Windows

**Operating system or device - Godot version:**
Windows 10, scons 2.4.1, MSVC 14, 89132224a651c0e0d4121270f63decb9a678ff88

**Issue description** (what happened, and what was expected):
```
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function register_module_types(): error: undefined reference to 'register_chibi_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function register_module_types(): error: undefined reference to 'register_cscript_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function register_module_types(): error: undefined reference to 'register_dds_types()'
... lots of "register_module_something()"

modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function unregister_module_types(): error: undefined reference to 'unregister_chibi_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function unregister_module_types(): error: undefined reference to 'unregister_cscript_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function unregister_module_types(): error: undefined reference to 'unregister_dds_types()'
... lots of "unregister_module_something()"
... and lots of "webp, VP8, TestGDScript::test" things

collect2.exe: error: ld returned 1 exit status

=====
scons: *** [bin\libgodot.android.opt.debug.armv7.neon.so] Error 1
scons: building terminated because of errors.
```
Compiling for windows tool and templates are fine.

**Steps to reproduce:**
`scons -j 4 platform=android colored=yes tools=no target=release_debug`

**Link to minimal example project** (optional but very welcome):

```
scons -j 4 platform=android tools=no target=release_debug
scons: Reading SConscript files ...
Godot Android!!!!! (armv7) (with neon)
scons: done reading SConscript files.
scons: Building targets ...
D:\Development\NVPACK\android-ndk-r12b/toolchains/arm-linux-androideabi-4.9/prebuilt//windows-x86_64/bin/arm-linux-androideabi-g++ -o core\script_encryption_key.android.opt.debug.armv7.neon.o -c -DNO_STATVFS -fpic -ffunction-sections -funwind-tables -fstack-protector -fvisibility=hidden -D__ARM_ARCH_7__ -D__ARM_ARCH_7A__ -D__GLIBC__ -Wno-psabi -march=armv7-a -mfloat-abi=softfp -ftree-vectorize -funsafe-math-optimizations -fno-strict-aliasing -DANDROID -Wa,--noexecstack -DGLES2_ENABLED -mfpu=neon -D__ARM_NEON__ -O2 -ffast-math -DDEBUG_ENABLED -fno-exceptions -DNO_SAFE_CAST -DFREETYPE_ENABLED -DENABLE_DEPRECATED -DANDROID_ENABLED -DUNIX_ENABLED -DNO_FCNTL -DMPC_FIXED_POINT -DGDSCRIPT_ENABLED -DMINIZIP_ENABLED -DXML_ENABLED -Icore -Icore\math -Itools -Idrivers -I. -Iplatform\android -ID:\Development\NVPACK\android-ndk-r12b\platforms\android-14\arch-arm\usr\include -ID:\Development\NVPACK\android-ndk-r12b\sources\cxx-stl\gnu-libstdc++\4.9\include -ID:\Development\NVPACK\android-ndk-r12b\sources\cpufeatures -Ithirdparty\freetype -Ithirdparty\freetype\include -Ithirdparty\libpng core\script_encryption_key.cpp
D:\Development\NVPACK\android-ndk-r12b/toolchains/arm-linux-androideabi-4.9/prebuilt//windows-x86_64/bin/arm-linux-androideabi-ar rc core\libcore.android.opt.debug.armv7.neon.a platform\android\globals\global_defaults.android.opt.debug.armv7.neon.o platform\iphone\globals\global_defaults.android.opt.debug.armv7.neon.o core\array.android.opt.debug.armv7.neon.o core\color.android.opt.debug.armv7.neon.o core\command_queue_mt.android.opt.debug.armv7.neon.o core\compressed_translation.android.opt.debug.armv7.neon.o core\core_string_names.android.opt.debug.armv7.neon.o core\dictionary.android.opt.debug.armv7.neon.o core\dvector.android.opt.debug.armv7.neon.o core\error_macros.android.opt.debug.armv7.neon.o core\event_queue.android.opt.debug.armv7.neon.o core\func_ref.android.opt.debug.armv7.neon.o core\globals.android.opt.debug.armv7.neon.o core\global_constants.android.opt.debug.armv7.neon.o core\global_defaults.android.opt.debug.armv7.neon.o core\hq2x.android.opt.debug.armv7.neon.o core\image.android.opt.debug.armv7.neon.o core\image_quantize.android.opt.debug.armv7.neon.o core\input_map.android.opt.debug.armv7.neon.o core\message_queue.android.opt.debug.armv7.neon.o core\method_bind.android.opt.debug.armv7.neon.o core\object.android.opt.debug.armv7.neon.o core\object_type_db.android.opt.debug.armv7.neon.o core\packed_data_container.android.opt.debug.armv7.neon.o core\path_db.android.opt.debug.armv7.neon.o core\path_remap.android.opt.debug.armv7.neon.o core\pool_allocator.android.opt.debug.armv7.neon.o core\print_string.android.opt.debug.armv7.neon.o core\profile_clock.android.opt.debug.armv7.neon.o core\reference.android.opt.debug.armv7.neon.o core\ref_ptr.android.opt.debug.armv7.neon.o core\register_core_types.android.opt.debug.armv7.neon.o core\resource.android.opt.debug.armv7.neon.o core\rid.android.opt.debug.armv7.neon.o core\safe_refcount.android.opt.debug.armv7.neon.o core\script_debugger_local.android.opt.debug.armv7.neon.o core\script_debugger_remote.android.opt.debug.armv7.neon.o core\script_encryption_key.android.opt.debug.armv7.neon.o core\script_language.android.opt.debug.armv7.neon.o core\string_db.android.opt.debug.armv7.neon.o core\translation.android.opt.debug.armv7.neon.o core\undo_redo.android.opt.debug.armv7.neon.o core\ustring.android.opt.debug.armv7.neon.o core\variant.android.opt.debug.armv7.neon.o core\variant_call.android.opt.debug.armv7.neon.o core\variant_construct_string.android.opt.debug.armv7.neon.o core\variant_op.android.opt.debug.armv7.neon.o core\variant_parser.android.opt.debug.armv7.neon.o core\os\copymem.android.opt.debug.armv7.neon.o core\os\dir_access.android.opt.debug.armv7.neon.o core\os\file_access.android.opt.debug.armv7.neon.o core\os\input.android.opt.debug.armv7.neon.o core\os\input_event.android.opt.debug.armv7.neon.o core\os\keyboard.android.opt.debug.armv7.neon.o core\os\main_loop.android.opt.debug.armv7.neon.o core\os\memory.android.opt.debug.armv7.neon.o core\os\memory_pool_dynamic.android.opt.debug.armv7.neon.o core\os\memory_pool_dynamic_prealloc.android.opt.debug.armv7.neon.o core\os\memory_pool_dynamic_static.android.opt.debug.armv7.neon.o core\os\memory_pool_static.android.opt.debug.armv7.neon.o core\os\mutex.android.opt.debug.armv7.neon.o core\os\os.android.opt.debug.armv7.neon.o core\os\semaphore.android.opt.debug.armv7.neon.o core\os\shell.android.opt.debug.armv7.neon.o core\os\thread.android.opt.debug.armv7.neon.o core\os\thread_dummy.android.opt.debug.armv7.neon.o core\os\thread_safe.android.opt.debug.armv7.neon.o core\math\aabb.android.opt.debug.armv7.neon.o core\math\a_star.android.opt.debug.armv7.neon.o core\math\bsp_tree.android.opt.debug.armv7.neon.o core\math\camera_matrix.android.opt.debug.armv7.neon.o core\math\face3.android.opt.debug.armv7.neon.o core\math\geometry.android.opt.debug.armv7.neon.o core\math\math_2d.android.opt.debug.armv7.neon.o core\math\math_funcs.android.opt.debug.armv7.neon.o core\math\matrix3.android.opt.debug.armv7.neon.o core\math\plane.android.opt.debug.armv7.neon.o core\math\quat.android.opt.debug.armv7.neon.o core\math\quick_hull.android.opt.debug.armv7.neon.o core\math\transform.android.opt.debug.armv7.neon.o core\math\triangle_mesh.android.opt.debug.armv7.neon.o core\math\triangulate.android.opt.debug.armv7.neon.o core\math\triangulator.android.opt.debug.armv7.neon.o core\math\vector3.android.opt.debug.armv7.neon.o core\io\aes256.android.opt.debug.armv7.neon.o core\io\compression.android.opt.debug.armv7.neon.o core\io\config_file.android.opt.debug.armv7.neon.o core\io\file_access_buffered.android.opt.debug.armv7.neon.o core\io\file_access_compressed.android.opt.debug.armv7.neon.o core\io\file_access_encrypted.android.opt.debug.armv7.neon.o core\io\file_access_memory.android.opt.debug.armv7.neon.o core\io\file_access_network.android.opt.debug.armv7.neon.o core\io\file_access_pack.android.opt.debug.armv7.neon.o core\io\file_access_zip.android.opt.debug.armv7.neon.o core\io\http_client.android.opt.debug.armv7.neon.o core\io\image_loader.android.opt.debug.armv7.neon.o core\io\ip.android.opt.debug.armv7.neon.o core\io\ip_address.android.opt.debug.armv7.neon.o core\io\json.android.opt.debug.armv7.neon.o core\io\marshalls.android.opt.debug.armv7.neon.o core\io\md5.android.opt.debug.armv7.neon.o core\io\networked_multiplayer_peer.android.opt.debug.armv7.neon.o core\io\packet_peer.android.opt.debug.armv7.neon.o core\io\packet_peer_udp.android.opt.debug.armv7.neon.o core\io\pck_packer.android.opt.debug.armv7.neon.o core\io\resource_format_binary.android.opt.debug.armv7.neon.o core\io\resource_format_xml.android.opt.debug.armv7.neon.o core\io\resource_loader.android.opt.debug.armv7.neon.o core\io\resource_saver.android.opt.debug.armv7.neon.o core\io\stream_peer.android.opt.debug.armv7.neon.o core\io\stream_peer_ssl.android.opt.debug.armv7.neon.o core\io\stream_peer_tcp.android.opt.debug.armv7.neon.o core\io\tcp_server.android.opt.debug.armv7.neon.o core\io\translation_loader_po.android.opt.debug.armv7.neon.o core\io\xml_parser.android.opt.debug.armv7.neon.o core\io\base64.android.opt.debug.armv7.neon.o core\io\fastlz.android.opt.debug.armv7.neon.o core\io\ioapi.android.opt.debug.armv7.neon.o core\io\sha256.android.opt.debug.armv7.neon.o core\io\unzip.android.opt.debug.armv7.neon.o core\io\zip.android.opt.debug.armv7.neon.o core\bind\core_bind.android.opt.debug.armv7.neon.o
D:\Development\NVPACK\android-ndk-r12b/toolchains/arm-linux-androideabi-4.9/prebuilt//windows-x86_64/bin/arm-linux-androideabi-ranlib core\libcore.android.opt.debug.armv7.neon.a
D:\Development\NVPACK\android-ndk-r12b/toolchains/arm-linux-androideabi-4.9/prebuilt//windows-x86_64/bin/arm-linux-androideabi-g++ -o bin\libgodot.android.opt.debug.armv7.neon.so -g --sysroot=D:\Development\NVPACK\android-ndk-r12b/platforms/android-14/arch-arm -Wl,--no-undefined -Wl,-z,noexecstack -Wl,-soname,libgodot_android.so -shared platform\android\os_android.os platform\android\godot_android.os platform\android\file_access_android.os platform\android\dir_access_android.os platform\android\audio_driver_opensl.os platform\android\file_access_jandroid.os platform\android\dir_access_jandroid.os platform\android\thread_jandroid.os platform\android\audio_driver_jandroid.os platform\android\ifaddrs_android.os platform\android\android_native_app_glue.os platform\android\java_glue.os platform\android\cpu-features.os platform\android\java_class_wrapper.os -LD:\Development\NVPACK\android-ndk-r12b\sources\cxx-stl\gnu-libstdc++\4.9\libs\armeabi-v7a main\libmain.android.opt.debug.armv7.neon.a modules\libmodules.android.opt.debug.armv7.neon.a bin\tests\libtests.android.opt.debug.armv7.neon.a drivers\libdrivers.android.opt.debug.armv7.neon.a drivers\libdrivers0.android.opt.debug.armv7.neon.a scene\libscene.android.opt.debug.armv7.neon.a servers\libservers.android.opt.debug.armv7.neon.a core\libcore.android.opt.debug.armv7.neon.a -lOpenSLES -lEGL -lOpenSLES -landroid -lc -lm -lstdc++ -llog -lGLESv1_CM -lGLESv2 -lz -lgnustl_static modules\freetype\libfreetype_builtin.android.opt.debug.armv7.neon.a
=====
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function register_module_types(): error: undefined reference to 'register_chibi_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function register_module_types(): error: undefined reference to 'register_cscript_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function register_module_types(): error: undefined reference to 'register_dds_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function register_module_types(): error: undefined reference to 'register_enet_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function register_module_types(): error: undefined reference to 'register_etc1_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function register_module_types(): error: undefined reference to 'register_freetype_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function register_module_types(): error: undefined reference to 'register_gdscript_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function register_module_types(): error: undefined reference to 'register_gridmap_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function register_module_types(): error: undefined reference to 'register_ik_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function register_module_types(): error: undefined reference to 'register_jpg_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function register_module_types(): error: undefined reference to 'register_mpc_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function register_module_types(): error: undefined reference to 'register_openssl_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function register_module_types(): error: undefined reference to 'register_opus_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function register_module_types(): error: undefined reference to 'register_pbm_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function register_module_types(): error: undefined reference to 'register_pvr_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function register_module_types(): error: undefined reference to 'register_theora_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function register_module_types(): error: undefined reference to 'register_visual_script_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function register_module_types(): error: undefined reference to 'register_vorbis_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function register_module_types(): error: undefined reference to 'register_webp_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function unregister_module_types(): error: undefined reference to 'unregister_chibi_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function unregister_module_types(): error: undefined reference to 'unregister_cscript_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function unregister_module_types(): error: undefined reference to 'unregister_dds_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function unregister_module_types(): error: undefined reference to 'unregister_enet_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function unregister_module_types(): error: undefined reference to 'unregister_etc1_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function unregister_module_types(): error: undefined reference to 'unregister_freetype_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function unregister_module_types(): error: undefined reference to 'unregister_gdscript_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function unregister_module_types(): error: undefined reference to 'unregister_gridmap_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function unregister_module_types(): error: undefined reference to 'unregister_ik_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function unregister_module_types(): error: undefined reference to 'unregister_jpg_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function unregister_module_types(): error: undefined reference to 'unregister_mpc_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function unregister_module_types(): error: undefined reference to 'unregister_openssl_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function unregister_module_types(): error: undefined reference to 'unregister_opus_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function unregister_module_types(): error: undefined reference to 'unregister_pbm_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function unregister_module_types(): error: undefined reference to 'unregister_pvr_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function unregister_module_types(): error: undefined reference to 'unregister_theora_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function unregister_module_types(): error: undefined reference to 'unregister_visual_script_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function unregister_module_types(): error: undefined reference to 'unregister_vorbis_types()'
modules\libmodules.android.opt.debug.armv7.neon.a(register_module_types.android.opt.debug.armv7.neon.o):register_module_types.cpp:function unregister_module_types(): error: undefined reference to 'unregister_webp_types()'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::_parser_expr(GDParser::Node const*): error: undefined reference to 'GDFunctions::get_func_name(GDFunctions::Function)'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::_disassemble_addr(Ref<GDScript> const&, GDFunction const&, int): error: undefined reference to 'GDScript::debug_get_member_by_index(int) const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::_disassemble_addr(Ref<GDScript> const&, GDFunction const&, int): error: undefined reference to 'GDFunction::get_global_name(int) const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::_disassemble_addr(Ref<GDScript> const&, GDFunction const&, int): error: undefined reference to 'GDFunction::get_constant(int) const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::_disassemble_addr(Ref<GDScript> const&, GDFunction const&, int): error: undefined reference to 'GDFunction::get_global_name(int) const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDTokenizerText::set_code(String const&)'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDTokenizerText::get_token(int) const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDTokenizerText::get_token(int) const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDTokenizerText::get_token(int) const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDTokenizerText::get_token(int) const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDTokenizer::get_token_name(GDTokenizer::Token)'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDTokenizerText::get_token_line(int) const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDTokenizerText::get_token_line(int) const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDTokenizerText::get_token_column(int) const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDTokenizerText::advance(int)'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDTokenizerText::get_token_identifier(int) const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDTokenizerText::get_token_constant(int) const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDTokenizerText::get_token_error(int) const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDTokenizerText::get_token_line(int) const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDTokenizerText::get_token_line_indent(int) const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDTokenizerBuffer::parse_code_string(String const&)'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDParser::GDParser()'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDParser::parse(String const&, String const&, bool, String const&, bool)'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDParser::get_parse_tree() const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDParser::~GDParser()'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'vtable for GDTokenizerText'
d:/development/nvpack/android-ndk-r12b/toolchains/arm-linux-androideabi-4.9/prebuilt/windows-x86_64/bin/../lib/gcc/arm-linux-androideabi/4.9.x/../../../../arm-linux-androideabi/bin/ld.exe: the vtable symbol may be undefined because the class is missing its key function (see go/missingkeymethod)
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDTokenizerText::get_token_built_in_func(int) const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDFunctions::get_func_name(GDFunctions::Function)'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDParser::GDParser()'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDParser::parse(String const&, String const&, bool, String const&, bool)'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDScript::GDScript()'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDCompiler::GDCompiler()'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDCompiler::compile(GDParser const*, GDScript*, bool)'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDScript::debug_get_member_functions() const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDFunction::get_code() const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDFunction::get_code_size() const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDFunction::get_default_argument_count() const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDFunction::get_name() const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDFunction::get_max_stack_size() const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDParser::get_error_line() const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDParser::get_error_column() const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDParser::get_error() const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDScript::get_base() const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDParser::~GDParser()'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDFunction::get_global_name(int) const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDFunctions::get_func_name(GDFunctions::Function)'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDFunction::get_global_name(int) const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDParser::get_error_line() const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDParser::get_error_column() const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDParser::get_error() const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDFunction::get_default_argument_addr(int) const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDFunction::get_default_argument_count() const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDParser::~GDParser()'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDCompiler::get_error_line() const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDCompiler::get_error_column() const'
bin\tests\libtests.android.opt.debug.armv7.neon.a(test_gdscript.android.opt.debug.armv7.neon.o):test_gdscript.cpp:function TestGDScript::test(TestGDScript::TestType): error: undefined reference to 'GDCompiler::get_error() const'
collect2.exe: error: ld returned 1 exit status

=====
scons: *** [bin\libgodot.android.opt.debug.armv7.neon.so] Error 1
scons: building terminated because of errors.
```

--- a/SConstruct
+++ b/SConstruct
@@ -100,6 +100,7 @@ env_base.__class__.disable_module = methods.disable_module
 
 env_base.__class__.add_source_files = methods.add_source_files
 env_base.__class__.use_windows_spawn_fix = methods.use_windows_spawn_fix
+env_base.__class__.split_lib = methods.split_lib
 
 env_base["x86_libtheora_opt_gcc"]=False
 env_base["x86_libtheora_opt_vc"]=False

--- a/drivers/SCsub
+++ b/drivers/SCsub
@@ -35,6 +35,9 @@ if (env["tools"]=="yes"):
 if env['vsproj']=="yes":
 	env.AddToVSProject(env.drivers_sources)
 
-env.add_source_files(env.drivers_sources,"*.cpp")
-lib = env.Library("drivers",env.drivers_sources)
-env.Prepend(LIBS=[lib])
+if env.split_drivers:
+	env.split_lib("drivers")
+else:
+	env.add_source_files(env.drivers_sources,"*.cpp")
+	lib = env.Library("drivers",env.drivers_sources)
+	env.Prepend(LIBS=[lib])

--- a/drivers/SCsub
+++ b/drivers/SCsub
@@ -35,52 +35,6 @@ if (env["tools"]=="yes"):
 if env['vsproj']=="yes":
 	env.AddToVSProject(env.drivers_sources)
 
-
-# Split drivers, this used to be needed for windows until separate builders for windows were created
-# FIXME: Check if still needed now that the drivers were made more lightweight
-if (env.split_drivers):
-	import string
-
-	num = 0
-	cur_base = ""
-	max_src = 64
-	list = []
-	lib_list = []
-
-	for f in env.drivers_sources:
-		fname = ""
-		if type(f) == type(""):
-			fname = env.File(f).path
-		else:
-			fname = env.File(f)[0].path
-		fname = fname.replace("\\", "/")
-		base = string.join(fname.split("/")[:2], "/")
-		if base != cur_base and len(list) > max_src:
-			if num > 0:
-				lib = env.Library("drivers"+str(num), list)
-				lib_list.append(lib)
-				list = []
-			num = num+1
-		cur_base = base
-		list.append(f)
-
-	lib = env.Library("drivers"+str(num), list)
-	lib_list.append(lib)
-
-	if len(lib_list) > 0:
-		import os, sys
-		if os.name=='posix' and sys.platform=='msys':
-			env.Replace(ARFLAGS=['rcsT'])
-
-			lib = env.Library("drivers_collated", lib_list)
-			lib_list = [lib]
-
-	drivers_base=[]
-	env.add_source_files(drivers_base,"*.cpp")
-	lib_list.insert(0, env.Library("drivers", drivers_base))
-
-	env.Prepend(LIBS=lib_list)
-else:
-	env.add_source_files(env.drivers_sources,"*.cpp")
-	lib = env.Library("drivers",env.drivers_sources)
-	env.Prepend(LIBS=[lib])
+env.add_source_files(env.drivers_sources,"*.cpp")
+lib = env.Library("drivers",env.drivers_sources)
+env.Prepend(LIBS=[lib])

--- a/methods.py
+++ b/methods.py
@@ -1352,7 +1352,15 @@ def use_windows_spawn_fix(self, platform=None):
     if (os.name!="nt"):
 	return #not needed, only for windows
 
-    self.split_drivers=True
+    # On Windows, due to the limited command line length, when creating a static library
+    # from a very high number of objects SCons will invoke "ar" once per object file;
+    # that makes object files with same names to be overwritten so the last wins and
+    # the library looses symbols defined by overwritten objects.
+    # By enabling quick append instead of the default mode (replacing), libraries will
+    # got built correctly regardless the invokation strategy.
+    # Furthermore, since SCons will rebuild the library from scratch when an object file
+    # changes, no multiple versions of the same object file will be present.
+    self.Replace(ARFLAGS='q')
 
     import subprocess
 

--- a/methods.py
+++ b/methods.py
@@ -1402,6 +1402,50 @@ def use_windows_spawn_fix(self, platform=None):
     self['SPAWN'] = mySpawn
 
 
+def split_lib(self, libname):
+	import string
+	env = self
+
+	num = 0
+	cur_base = ""
+	max_src = 64
+	list = []
+	lib_list = []
+
+	for f in getattr(env, libname + "_sources"):
+		fname = ""
+		if type(f) == type(""):
+			fname = env.File(f).path
+		else:
+			fname = env.File(f)[0].path
+		fname = fname.replace("\\", "/")
+		base = string.join(fname.split("/")[:2], "/")
+		if base != cur_base and len(list) > max_src:
+			if num > 0:
+				lib = env.Library(libname + str(num), list)
+				lib_list.append(lib)
+				list = []
+			num = num + 1
+		cur_base = base
+		list.append(f)
+
+	lib = env.Library(libname + str(num), list)
+	lib_list.append(lib)
+
+	if len(lib_list) > 0:
+		import os, sys
+		if os.name == 'posix' and sys.platform == 'msys':
+			env.Replace(ARFLAGS = ['rcsT'])
+			lib = env.Library(libname + "_collated", lib_list)
+			lib_list = [lib]
+
+	lib_base = []
+	env.add_source_files(lib_base, "*.cpp")
+	lib_list.insert(0, env.Library(libname, lib_base))
+
+	env.Prepend(LIBS = lib_list)
+
+
 def save_active_platforms(apnames,ap):
 
 	for x in ap:

